mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/dao/PostgresMessageDAO.java

```diff
@@ -55,11 +57,27 @@
 import reactor.core.scheduler.Schedulers;
 
 public class PostgresMessageDAO {
+
+    public static class Factory {
+        private final BlobId.Factory blobIdFactory;
+        private final PostgresExecutor.Factory executorFactory;
+
+        @Inject
+        @Singleton
+        public Factory(BlobId.Factory blobIdFactory, PostgresExecutor.Factory executorFactory) {
+            this.blobIdFactory = blobIdFactory;
+            this.executorFactory = executorFactory;
+        }
+
+        public PostgresMessageDAO create(Optional<Domain> domain) {
+            return new PostgresMessageDAO(executorFactory.create(domain), blobIdFactory);
+        }
+    }
+
     public static final long DEFAULT_LONG_VALUE = 0L;
     private final PostgresExecutor postgresExecutor;
```

Comment:
why not using PostgresExecutor.Factory here instead of creating a new factory for PostgresMessageDAO?